src/tools/rbd_mirror/ImageReplayer.cc

```diff
@@ -1437,6 +1468,25 @@ void ImageReplayer<I>::handle_shut_down(int r, Context *on_start) {
 }
 
 template <typename I>
+void ImageReplayer<I>::handle_remote_journal_metadata_updated() {
+  dout(20) << dendl;
+
+  cls::journal::Client client;
+  int r = m_remote_journaler->get_cached_client(m_local_mirror_uuid, &client);
+  if (r < 0) {
+    derr << "failed to retrieve client: " << cpp_strerror(r) << dendl;
+    return;
+  }
+
+  dout(20) << "client=" << client << dendl;
+
+  if (client.state != cls::journal::CLIENT_STATE_CONNECTED) {
+    dout(0) << "client flagged disconnected, stopping image replay" << dendl;
+    stop(); // XXX: Schedule re-sync after configurable interval?
```

Comment:
Should this stop with an error code so that ``` rbd mirror image status``` reports that it has been disconnected?